nixos/modules/services/hardware/nvidia-container-toolkit/cdi-generate.nix

```diff
@@ -0,0 +1,37 @@
+{ config, pkgs }: let
+  mountOptions = { options = ["ro" "nosuid" "nodev" "bind"]; };
+  mounts = [
+    { hostPath = "${config.boot.kernelPackages.nvidia_x11}/bin/nvidia-cuda-mps-control";
+      containerPath = "/usr/bin/nvidia-cuda-mps-control"; }
+    { hostPath = "${config.boot.kernelPackages.nvidia_x11}/bin/nvidia-cuda-mps-server";
+      containerPath = "/usr/bin/nvidia-cuda-mps-server"; }
+    { hostPath = "${config.boot.kernelPackages.nvidia_x11}/bin/nvidia-debugdump";
+      containerPath = "/usr/bin/nvidia-debugdump"; }
+    { hostPath = "${config.boot.kernelPackages.nvidia_x11}/bin/nvidia-powerd";
+      containerPath = "/usr/bin/nvidia-powerd"; }
+    { hostPath = "${config.boot.kernelPackages.nvidia_x11}/bin/nvidia-smi";
+      containerPath = "/usr/bin/nvidia-smi"; }
+    { hostPath = "${pkgs.nvidia-container-toolkit}/bin/nvidia-ctk";
+      containerPath = "/usr/bin/nvidia-ctk"; }
+    { hostPath = "${pkgs.glibc}/lib";
+      containerPath = "${pkgs.glibc}/lib"; }
```

Comment:
@SomeoneSerge should I do `/run/booted-system/bin/` instead? If a new NixOS configuration is activated, I think this will service will be restarted, pointing to the newly installed `nvidia_x11` instead of the booted one.

I think it would be better to point to `/run/booted-system` (didn't know about this one, thanks for the pointer). What do you think?

Reply:
The issue, unfortunately, is more delicate, cf. the quotations from K900 in https://github.com/NixOS/nixpkgs/issues/255070. TLDR: we might want to do `booted-system` for `libcuda` and `current-system` for `*GL`. I feel like this is complex enough that we should reserve this for a new PR

Reply:
Oh, I see. Thanks for the context. We can work on this on a different PR for sure.

Reply:
A different comment though: the user might choose `nvidia_x11_beta` (or any of the other options) instead of `nvidia_x11`. So `hardware.nvidia.package` again, iirc

Reply:
I just moved everything to `booted-system`.

Reply:
>  also note that the booted-system might not have had nvidia_x11 deployed at all (we boot into a system without any drivers, then we nixos-rebuild switch into a system with xserver.videoDrivers = [ "nvidia" ])

Reply:
Yes, either we have to choose that, or things breaking from time to time as far as I understood. I don't have a preference, I just thought that things breaking from time to time was a worse experience than things being broke the very first time.

Reply:
I think things "breaking from time to time" is really "needing to reboot after updating the kernel (modules)" and closely related to the mentioned libcuda/mesa issue. These issues usually present themselves in the form of error messages about driver versions. On the other hand, having to reboot the first time you enable `hardware.opengl.enable` or, in this case, `nvidia-container-toolkit` is maybe less intuitive. The errors we'd see would be about missing files, which are infrequent on NixOS.

TLDR: I think it'd be more consistent to use paths that are guaranteed to exist, than to use potentially missing paths that "we believe are more likely to work after upgrades", which belief actually requires a bit more research

Reply:
And if using `current-system` instead of store paths... those are deployed by `nixos/modules/hardware/video/nvidia.nix` I think? We should ideally add an assertion for `nvidiaEnabled || datacenter.enable where nvidiaEnabled = lib.elem "nvidia" config.services.xserver.videoDrivers`